src/websocket.c

```diff
@@ -387,21 +513,92 @@ ws_reply(struct cmd *cmd, const char *p, size_t sz) {
 		frame_sz = sz + 10;
 	}
 
-	/* send WS frame */
-	r = http_response_init(cmd->w, 0, NULL);
-	if (r == NULL) {
-		free(frame);
-		slog(cmd->w->s, WEBDIS_ERROR, "Failed response allocation in ws_reply", 0);
+	/* mark as keep alive, otherwise we'll close the connection after the first reply */
+	int add_ret = evbuffer_add(ws->wbuf, frame, frame_sz);
+	free(frame); /* no longer needed once added to buffer */
+	if(add_ret < 0) {
+		slog(ws->http_client->w->s, WEBDIS_ERROR, "Failed response allocation in ws_frame_and_send_response", 0);
 		return -1;
 	}
 
-	/* mark as keep alive, otherwise we'll close the connection after the first reply */
-	r->keep_alive = 1;
-
-	r->out = frame;
-	r->out_sz = frame_sz;
-	r->sent = 0;
-	http_schedule_write(cmd->fd, r);
+	/* send WS frame */
+	ws_schedule_write(ws);
 
 	return 0;
 }
+
+static void
+ws_close_if_able(struct ws_client *ws) {
```

Comment:
very minor, formatting: most if not all other functions start with a new line.  Sorry this feels like nitpicking and I don't really care either way as long as it's consistent.